homeworks/rostyslav.havuka_RostyslavHavuka/6-typeScript/src/index.ts

```diff
@@ -0,0 +1,247 @@
+      alert('Please press "Start"')
+    }
+  }
+
+  private falseGame(): void {
+    if (this.gameState === true) {
+      this.subScore()
+      this.startKeysInterval()
+      this.setNewInterval()
+      this.sizeDown()
+      clearInterval(this.intervalStatusBarId)
+      this.width = 1;
+      this.timerProgressBar = 0;
+      this.progressBar()
+    } else {
+      alert('Please press "Start"')
+    }
+  }
+
+  private setKey(e: any): void {
+    if (e.key === keyElement.value.toLowerCase()) {
+      this.trueGame()
+    }
+    else {
+      this.falseGame()
+    }
+  }
+
+  private setNewInterval(): void {
+    if (this.gameState === true) {
+      clearInterval(this.intervalId)
+      this.intervalId = setInterval(() => {
+        this.subScoreTimer(this.score)
+        this.sizeDown()
+        clearInterval(this.intervalStatusBarId)
+        this.width = 1;
+        this.timerProgressBar = 0;
+        this.progressBar()
+      }, 3000);
+    }
+  }
+
+  private addScore(): void {
+    let rndom: number = this.randomNumber(5, 10)
+    this.score = this.score + rndom
+    cubeScoreElement.innerText = this.score.toString()
+    scoreElement.innerText = `+${rndom}`
+  }
+
+  private subScore(): void {
+    let rndom: number = this.randomNumber(20, 25)
+    this.score = this.score - rndom
+    cubeScoreElement.innerText = this.score.toString()
+    scoreElement.innerText = `-${rndom}`
+  }
+
+  private subScoreTimer(score: number): void {
+    let rndom: number = this.randomNumber(10, 15)
+    this.score = this.score - rndom
+    this.finishGame()
+    cubeScoreElement.innerText = this.score.toString()
+    scoreElement.innerText = `-${rndom}`
+  }
+
+  private subscribeOnKeyPress(): void {
+    window.addEventListener('keypress', (e) => {
+      this.setKey(e)
+      this.finishGame()
+    })
+  }
+
+  private randomNumber(min: number, max: number) {
+    return Math.floor(Math.random() * (max - min + 1)) + min;
+  }
+
+  private sizeUp(): void {
+    let obj: HTMLElement = document.getElementById("box");
+    let width: number = parseInt(obj.style.width);
+    let height: number = parseInt(obj.style.height);
+    obj.style.width = (width + 3) + "px";
+    obj.style.height = (height + 3) + "px";
+  }
+
+  private sizeDown(): void {
+    let obj: HTMLElement = document.getElementById("box");
+    let width: number = parseInt(obj.style.width);
+    let height: number = parseInt(obj.style.height);
+    obj.style.width = (width - 2) + "px";
+    obj.style.height = (height - 2) + "px";
+  }
+
+  private progressBar(): void {
+    if (this.timerProgressBar == 0) {
+      this.timerProgressBar = 1;
+      const element: any = document.getElementsByClassName('myBar');
+      this.width = 1;
+      this.intervalStatusBarId = setInterval(() => {
+        if (this.width >= 100) {
+          clearInterval(this.intervalStatusBarId)
+          this.timerProgressBar = 0;
+        } else {
+          this.width++;
+          for (const elem of element) {
```

Comment:
That's how you get the bar child element? 😲
This is a really bad solution. Pass it through the constructor